modules/deploy/data/cml.sh

```diff
@@ -9,91 +9,130 @@
+
+    # We need to wait until the initial setup is done
+    attempts=5
+    while [ $attempts -gt 0 ]; do
+        if systemctl show virl2-initial-setup.service | grep -qi 'SubState=exited'; then
+            echo "initial setup is done"
+            break
+        fi
+        echo "waiting for initial setup..."
+        sleep 5
+        ((attempts--))
+    done
 
-    # if there's no images at this point, copy what's available in the defined
-    # cloud storage container
-    if [ $(find $VLLI -type f | wc -l) -eq 0 ]; then
-        copyfile refplat/ $VLLI/ "" --recursive
+    if [ $attempts -eq 0 ]; then
+        echo "initial setup did not finish in time... something went wrong!"
+        exit 1
     fi
 
-    systemctl stop ssh
-    apt-get install -y /provision/${CFG_APP_DEB}
-    systemctl start ssh
+    # for good measure, apply the network config again
+    netplan apply
+    systemctl enable --now ssh.service
+
+    # clean up software .pkg / .deb packages
+    rm -f /provision/*.pkg /provision/*.deb /tmp/*.deb
+
+    # enable and configure PaTTY
+    if [ "${CFG_COMMON_ENABLE_PATTY}" = "true" ]; then
+        GWDEV=$(ip -json route | jq -r '.[]|select(.dst=="default")|.dev')
+        echo "OPTS=\"-bridge $GWDEV -poll 5\"" >>/etc/default/patty.env
+        sed -i '/^After/iWants=virl2-patty.service' /lib/systemd/system/virl2.target
+        systemctl daemon-reload
+        systemctl enable --now virl2-patty
+    fi
 
     # AWS specific (?):
     # For troubleshooting. To allow console access on AWS, the root user needs a
     # password. Note: not all instance types / flavors provide a serial console!
     # echo "root:secret-password-here" | /usr/sbin/chpasswd
 }
 
-
 function cml_configure() {
     target=$1
+    CONFIG_FILE="/etc/virl2-base-config.yml"
     API="http://ip6-localhost:8001/api/v0"
 
-    # Create system user
-    /usr/sbin/useradd --badname -m -s /bin/bash ${CFG_SYS_USER}
-    echo "${CFG_SYS_USER}:${CFG_SYS_PASS}" | /usr/sbin/chpasswd
-    /usr/sbin/usermod -a -G sudo ${CFG_SYS_USER}
-
-    # Move SSH config from default cloud-provisioned user to new user. This
-    # also disables the login for this user by removing the SSH key.
-    # Technically, this could be the same user as Azure allows to set the
-    # user name
-    # if [ "$target" = "aws" ]; then
-    #     clouduser="ubuntu"
-    # elif [ "$target" = "azure" ]; then
-    #     clouduser="adminuser"
-    # else
-    #     echo "unknown target"
-    # fi
     clouduser="ubuntu"
-    mv /home/$clouduser/.ssh /home/${CFG_SYS_USER}/
+    if [[ -d /home/${CFG_SYS_USER}/.ssh ]]; then
+        # Directory exists - Move individual files within .ssh
+        mv /home/$clouduser/.ssh/* /home/${CFG_SYS_USER}/.ssh/
+    else
+        # Directory doesn't exist - Move the entire .ssh directory
+        mv /home/$clouduser/.ssh/ /home/${CFG_SYS_USER}/
+    fi
     chown -R ${CFG_SYS_USER}.${CFG_SYS_USER} /home/${CFG_SYS_USER}/.ssh
+    userdel -r $clouduser
```

Comment:
@rschmied, as we discussed yesterday, because of this line, the cloud-init job will fail:

```
2024-04-16 14:40:40,885 - handlers.py[DEBUG]: finish: modules-final/config-ssh_authkey_fingerprints: FAIL: running config-ssh_authkey_fingerprints with frequency once-per-instance
2024-04-16 14:40:40,959 - handlers.py[DEBUG]: finish: modules-final: FAIL: running modules for final 
```

```python
2024-04-16 14:40:40,885 - util.py[WARNING]: Running module ssh_authkey_fingerprints (<module 'cloudinit.config.cc_ssh_authkey_fingerprints' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_ssh_authkey_fingerprints.py'>) failed
2024-04-16 14:40:40,885 - util.py[DEBUG]: Running module ssh_authkey_fingerprints (<module 'cloudinit.config.cc_ssh_authkey_fingerprints' from '/usr/lib/python3/dist-packages/cloudinit/config/cc_ssh_authkey_fingerprints.py'>) failed
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cloudinit/config/modules.py", line 255, in _run_modules
    ran, _r = cc.run(
  File "/usr/lib/python3/dist-packages/cloudinit/cloud.py", line 60, in run
    return self._runners.run(name, functor, args, freq, clear_on_fail)
  File "/usr/lib/python3/dist-packages/cloudinit/helpers.py", line 172, in run
    results = functor(**args)
  File "/usr/lib/python3/dist-packages/cloudinit/config/cc_ssh_authkey_fingerprints.py", line 138, in handle
    (key_fn, key_entries) = ssh_util.extract_authorized_keys(user_name)
  File "/usr/lib/python3/dist-packages/cloudinit/ssh_util.py", line 410, in extract_authorized_keys
    (ssh_dir, pw_ent) = users_ssh_info(username)
  File "/usr/lib/python3/dist-packages/cloudinit/ssh_util.py", line 234, in users_ssh_info
    pw_ent = pwd.getpwnam(username)
KeyError: "getpwnam(): name not found: 'ubuntu'
```

Today, I tested deployment without deleting the `ubuntu` user, and the issue was not triggered.

The issue is cosmetic, but it may lead to some confusion. I guess it should be mentioned in the documentation if we want to keep it. 
I added it to clean the VM config. However, we can just block the user with `usermod -L.`